src/CMakeLists.txt

```diff
@@ -238,7 +238,7 @@ endif()
 target_compile_definitions(${src_objlib} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
 target_include_directories(${src_objlib} PRIVATE 
 		${CORE_LIB_INCLUDES} 
-		${EXTERNAL_INCLUDES}
+		SYSTEM ${EXTERNAL_INCLUDES}
```

Comment:
CC @dkeeney I'd like to include all external headers as `-isystem` instead of `-I`, which should avoid the problem (a warning in external header, which translates to error due to our strict -Werror) but it does not seem to work

Reply:
> using the SYSTEM keyword in target_include_directories 

I tried passing SYSTEM from cmake, `isystem` used to work in dark ages before we used cmake. 

> Is your objective to tell the compiler to treat all external includes as system so they don't get the warnings

yes :+1: 
esp. for "header-only" (as Eigen) that don't get built in a separate target, but are a header in the "main" build.